src/TestSuite/TestCase.php

```diff
@@ -856,12 +856,16 @@ public function getMockForModel(string $alias, array $methods = [], array $optio
         $existingMethods = array_intersect($methods, get_class_methods($className));
         $nonExistingMethods = array_diff($methods, $existingMethods);
 
-        /** @var \Cake\ORM\Table $mock */
-        $mock = $this->getMockBuilder($className)
+        $builder = $this->getMockBuilder($className)
             ->onlyMethods($existingMethods)
-            ->addMethods($nonExistingMethods)
-            ->setConstructorArgs([$options])
-            ->getMock();
+            ->setConstructorArgs([$options]);
+
+        if (!empty($nonExistingMethods)) {
```

Comment:
If ($...) 
The cloaking check is not necessary

Reply:
Sorry, I don't understand "the cloaking check", is the `!empty` part?